components/Login/Login.js

```diff
@@ -0,0 +1,181 @@
+import { useSession, signIn, signOut } from "next-auth/react";
+import LoginIcon from "@/public/icons/LoginIcon.svg";
+import LogoutIcon from "@/public/icons/LogoutIcon.svg";
+import styled from "styled-components";
+import { useEffect, useState } from "react";
+import RegularButton from "../Buttons/RegularButton";
+import ButtonWrapper from "../Buttons/ButtonWrapper";
+
+export default function Login({
+  position,
+  changeShowLogOutDialog,
+  showLogOutDialog,
+}) {
+  const { data: session } = useSession();
+
+  const [isSigning, setIsSigning] = useState("");
+
+  function handleLogout() {
+    signOut();
+    changeShowLogOutDialog(false);
+  }
+
+  useEffect(() => {
+    if (isSigning === "in") {
+      handleTimedOutSignIn();
+    }
+
+    function handleTimedOutSignIn() {
+      const signInTimeout = setTimeout(() => {
+        signIn();
+        setIsSigning("");
+      }, 300);
+
+      return () => {
+        clearTimeout(signInTimeout);
+      };
+    }
```

Comment:
this code does not work properly. The cleanup function with clearTimeout needs to be returned from the useEffect callback. Here it is returned from the internal handler function and is returned in line 25. It is not used so nothing will happen with it. You don't need the internal function, please remove the handler function and write the logic for the timeout directly in the callback. 
```suggestion
    if (isSigning !== "in") {
      return
    }

    const signInTimeout = setTimeout(() => {
      signIn();
      setIsSigning("");
    }, 300);

    return () => clearTimeout(signInTimeout);
```